utils/vlog/vlog_to_pbtype.py

```diff
@@ -179,26 +185,51 @@ def make_direct_conn(ic_xml, src, dst):
     # List of entries in format ((from_cell, from_pin), (to_cell, to_pin))
     interconn = []
 
+    # Determine multiple instances of the same cell:
+    cells = dict()
+    for cname, i_of in mod.cells:
+        if i_of in cells:
+            cells[i_of]['count'] += 1
+            cells[i_of]['is_multi_instance'] = True
+            # assign unique instance number
+            cells[i_of][cname] = cells[i_of]['count']
+        else:
+            cells[i_of] = dict()
+            cells[i_of]['is_multi_instance'] = False
+            cells[i_of]['count'] = 0
+            cells[i_of][cname] = 0
+
     # Blackbox modules don't have inner cells or interconnect (but do still have timing)
     if (not is_blackbox) or is_submode:
         # Process cells. First build the list of cnames.
+        processed_cells = list()
         for cname, i_of in mod.cells:
             pb_name = strip_name(i_of)
+            do_not_include = False
+            if i_of in processed_cells:
+                # do not emit xml include for every instance of multi instace cell
+                do_not_include = True
+            else:
+                processed_cells.append(i_of)
+            instance = -1
```

Comment:
worth adding a comment about the `instance` meaning?

Reply:
I added some comments